### PR TITLE
Fix hsplit vsplit dsplit crash when section is 0

### LINKED ISSUE #69270: torch.{h, v, d}split crash when sections = 0
## 🐛 Bug

It is similar with  #68970 . When sections = 0. `torch.hsplit(tensor, sections)` and `torch.vsplit(tensor, sections)` will crash (floating point exception)

## To Reproduce

```python
import torch
tensor = torch.rand(torch.Size([2, 2, 4]), dtype=torch.float32)
sections = 0
res1 = torch.hsplit(tensor, sections)
```
```python
import torch
tensor = torch.rand(torch.Size([2, 2, 4]), dtype=torch.float32)
sections = 0
res1 = torch.vsplit(tensor, sections)
```

floating point exception

## Expected behavior

They should not crash.

## Environment

- PyTorch Version (e.g., 1.0): 1.10.0, 1.9.0
- OS (e.g., Linux): Linux
- How you installed PyTorch (conda, pip, source): conda
- Python version: 3.9.5
- CUDA/cuDNN version: 11.1
- GPU models and configuration: RTX 3090

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser

### PR DESCRIPTION
**Pull request by xcnick**
Fixes #69270 #68970

Dim modulo by zero when split_size is 0.

### CODE REVIEW

**Comment by mruberry** [comment]
This looks great, @xcnick! Would you also add a test for this new behavior by adding ErrorInputs to the OpInfos for these operators? 

Take a look here for some sample error inputs:

https://github.com/pytorch/pytorch/blob/c236247826bbf49d4f491d97bac6df1da6c1abe8/torch/testing/_internal/common_methods_invocations.py#L6012

See the OpInfo note for details on OpInfos.

If adding the tests is too confusing or time consuming just let me know and we'll get them added later

**Comment by xcnick** [comment]
> This looks great, @xcnick! Would you also add a test for this new behavior by adding ErrorInputs to the OpInfos for these operators?
> 
> Take a look here for some sample error inputs:
> 
> https://github.com/pytorch/pytorch/blob/c236247826bbf49d4f491d97bac6df1da6c1abe8/torch/testing/_internal/common_methods_invocations.py#L6012
> 
> See the OpInfo note for details on OpInfos.
> 
> If adding the tests is too confusing or time consuming just let me know and we'll get them added later

These have been completed as requested.

**Comment by mruberry** [comment]
Thanks @xcnick! This looks great and the test failures are unrelated (sorry the CI is a little noisy).

I have one last question: why didn't tensor_split throw a runtime error properly and segfault instead?

https://github.com/pytorch/pytorch/blob/102684b2520de2b59f4e6162903ce50328def02d/aten/src/ATen/native/TensorShape.cpp#L560

The code looks like it should have detected this case, and when called from Python tensor_split doesn't crash when `sections=0`. Do you have an idea about what might be happening?

**Comment by xcnick** [comment]
> Thanks @xcnick! This looks great and the test failures are unrelated (sorry the CI is a little noisy).
> 
> I have one last question: why didn't tensor_split throw a runtime error properly and segfault instead?
> 
> https://github.com/pytorch/pytorch/blob/102684b2520de2b59f4e6162903ce50328def02d/aten/src/ATen/native/TensorShape.cpp#L560
> 
> The code looks like it should have detected this case, and when called from Python tensor_split doesn't crash when `sections=0`. Do you have an idea about what might be happening?

Well, I've tested tensor_split, and it throw a runtime error without segfault. The code you mentioned above ensures that there is no division by 0.

**Review by mruberry — APPROVED** [review]
Thanks @xcnick! Sorry this wasn't merged before the holidays, but I'll merge it now.
